x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/analytics_list/analytics_list.tsx

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, SFC, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  // EuiBadge,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiEmptyPrompt,
+  SortDirection,
+} from '@elastic/eui';
+
+import {
+  DataFrameAnalyticsId,
+  moveToAnalyticsWizard,
+  useRefreshAnalyticsList,
+} from '../../../../common';
+import { checkPermission } from '../../../../../privilege/check_privilege';
+import { getTaskStateBadge } from './columns';
+
+import {
+  DataFrameAnalyticsListColumn,
+  DataFrameAnalyticsListRow,
+  ItemIdToExpandedRowMap,
+  DATA_FRAME_TASK_STATE,
+  // DATA_FRAME_MODE,
+  Query,
+  Clause,
+} from './common';
+import { getAnalyticsFactory } from '../../services/analytics_service';
+import { getColumns } from './columns';
+import { ExpandedRow } from './expanded_row';
+import { ProgressBar, AnalyticsTable } from './analytics_table';
+import { useRefreshInterval } from './use_refresh_interval';
+
+function getItemIdToExpandedRowMap(
+  itemIds: DataFrameAnalyticsId[],
+  dataFrameAnalytics: DataFrameAnalyticsListRow[]
+): ItemIdToExpandedRowMap {
+  return itemIds.reduce(
+    (m: ItemIdToExpandedRowMap, analyticsId: DataFrameAnalyticsId) => {
+      const item = dataFrameAnalytics.find(analytics => analytics.config.id === analyticsId);
+      if (item !== undefined) {
+        m[analyticsId] = <ExpandedRow item={item} />;
+      }
+      return m;
+    },
+    {} as ItemIdToExpandedRowMap
+  );
+}
+
+function stringMatch(str: string | undefined, substr: string) {
```

Comment:
Looks like this function has been created and used in a couple of other places already - `transform_list.tsx` and `jobs_list/components/utils.js` - might be worth moving to a common place.

Reply:
Added a note to the [project](https://github.com/elastic/kibana/projects/36) to do this in a follow up (once we have a decision where transforms/analytics code will live eventually), I intentionally didn't do any code deduplication in this PR.